src/elf/program_header.rs

```diff
@@ -314,6 +314,9 @@ macro_rules! elf_program_header_std_impl { ($size:ty) => {
                 phdrs
             }
 
+            /// # Safety
+            ///
+            /// This function creates a `ProgramHeader` directly from a raw pointer
```

Comment:
nit: this should probably a note that the phnum value is blindly trusted and not validated in anyway (though not sure how you could)? but also maybe people should already know that if they're using this? dunno